src/JointAxis.cc

```diff
@@ -374,37 +375,54 @@ sdf::ElementPtr JointAxis::Element() const
   return this->dataPtr->sdf;
 }
 
-/////////////////////////////////////////////////
 sdf::ElementPtr JointAxis::ToElement(unsigned int _index) const
+{
+  sdf::Errors errors;
+  auto result = this->ToElement(errors, _index);
+  sdf::throwOrPrintErrors(errors);
+  return result;
+}
+
+/////////////////////////////////////////////////
+sdf::ElementPtr JointAxis::ToElement(sdf::Errors &_errors,
+                                     unsigned int _index) const
 {
   sdf::ElementPtr elem(new sdf::Element);
   sdf::initFile("joint.sdf", elem);
 
   std::string axisElemName = "axis";
   if (_index > 0u)
     axisElemName += std::to_string(_index + 1);
-  sdf::ElementPtr axisElem = elem->GetElement(axisElemName);
-  sdf::ElementPtr xyzElem = axisElem->GetElement("xyz");
-  xyzElem->Set<gz::math::Vector3d>(this->Xyz());
+  sdf::ElementPtr axisElem = elem->GetElement(axisElemName, _errors);
+  sdf::ElementPtr xyzElem = axisElem->GetElement("xyz", _errors);
+  xyzElem->Set<gz::math::Vector3d>(_errors, this->Xyz());
   if (!this->XyzExpressedIn().empty())
   {
     xyzElem->GetAttribute("expressed_in")->Set<std::string>(
-        this->XyzExpressedIn());
+        this->XyzExpressedIn(), _errors);
   }
-  sdf::ElementPtr dynElem = axisElem->GetElement("dynamics");
-  dynElem->GetElement("damping")->Set<double>(this->Damping());
-  dynElem->GetElement("friction")->Set<double>(this->Friction());
-  dynElem->GetElement("spring_reference")->Set<double>(
-      this->SpringReference());
-  dynElem->GetElement("spring_stiffness")->Set<double>(
-      this->SpringStiffness());
-
-  sdf::ElementPtr limitElem = axisElem->GetElement("limit");
-  limitElem->GetElement("lower")->Set<double>(this->Lower());
-  limitElem->GetElement("upper")->Set<double>(this->Upper());
-  limitElem->GetElement("effort")->Set<double>(this->Effort());
-  limitElem->GetElement("velocity")->Set<double>(this->MaxVelocity());
-  limitElem->GetElement("stiffness")->Set<double>(this->Stiffness());
-  limitElem->GetElement("dissipation")->Set<double>(this->Dissipation());
+  sdf::ElementPtr dynElem = axisElem->GetElement("dynamics", _errors);
+  dynElem->GetElement("damping", _errors)->Set<double>(
+      this->Damping());
```

Comment:
Sorry I didn't notice this last review, I think you might have missed using the new overloaded function with errors here

Reply:
Good catch, thanks!